incubator/hnc/internal/reconcilers/hnc_config.go

```diff
@@ -162,6 +163,22 @@ func (r *ConfigReconciler) validateRBACTypes(inst *api.HNCConfiguration) {
 	}
 }
 
+// validateModes set modes to ignore if they are invalid in the Spec. This func
+// can only be tested manually since the schema validation is now in place.
+// There's no way to add an "existing" config with unknown modes in the test.
+// TODO delete this function when v1alpha1 is no longer the storage version of
+//  HNCConfig CRD - https://github.com/kubernetes-sigs/multi-tenancy/issues/901
+func (r *ConfigReconciler) validateModes(inst *api.HNCConfiguration) {
+	for i := 0; i < len(inst.Spec.Types); i++ {
+		t := &inst.Spec.Types[i]
+		switch t.Mode {
+		case api.Propagate, api.Ignore, api.Remove, "":
```

Comment:
You don't want to change "" to Propagate? Or do we even need this function anymore?

Reply:
This field is still remain optional, so we can keep it as is for now?

Reply:
I'm not sure I understand. With the schema in place, what does this function do? The comment to the function says it can't be tested anymore, so why keep it around?

Reply:
hmm, this is for all other non-invalid mode to be set to `ignore` in line 176. If you are asking what `validateModes` func for, as stated in #899 , this is to fix any unknown modes in the already existing config resource so that the resource can be updated (otherwise with schema validation and existing invalid values, the resource cannot be updated).

Reply:
I wouldn't worry about fixing up existing specs as long as HNC doesn't do anything too terrible if bad values are already there - and even if they do, we can actually just ask people to fix it in the release notes.

Reply:
sgtm will remove this func in the reconciler.

Reply:
It actually affects user experience. Say I have a Secret with unknown mode and it is treated as `ignore` now. After using schema validation without this `validateModes` func, if users want to set type for ConfigMap with: 
```
k hns config set-type --apiVersion v1 --kind ConfigMap remove
```
They are not able to and will get this message that is not relevant to their changes:
```
Could not update the HNC Configuration: HNCConfiguration.hnc.x-k8s.io "config" is invalid: spec.types.mode: Unsupported value: "ignsssore": supported values: "propagate", "ignore", "remove"
```

Reply:
Yes, but I think that's fine given the circumstances (not too many users yet, beta software, etc). We can just put it in the release notes that the admin will need to fix any existing problems before they make any unrelated changes.

Reply:
ok sg makes sense let me remove it.